Pdf/PdfDocumentGenerator.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+using DocumentService.Pdf.Models;
+using System.Diagnostics;
+
+namespace DocumentService.Pdf
+{
+    public class FilePathDoesNotExist : Exception
+    {
+        public FilePathDoesNotExist(string msg) : base(msg)
+        {
+
+        }
+    }
+     
+    public class PdfDocumentGenerator
+    {
+        public static void generatePdfByTemplate(string templatePath, List<ContentMetaData> metaDataList, string outputFilePath)
+        {
+            try 
+            { 
+                if(!File.Exists(templatePath))
+                {
+                    throw new FilePathDoesNotExist("The file path you provided is not Valid"); 
+                }
+                
+                string modifiedHtmlFilePath =  ReplaceFileElementsWithMetaData(templatePath, metaDataList);
+                
+                ConversionCmd(modifiedHtmlFilePath, outputFilePath);
+
+            } 
+            catch(Exception e)
+            {
+                throw;
+            }
+        }
+
+        private static string ReplaceFileElementsWithMetaData(string templatePath, List<ContentMetaData> metaDataList)
+        {
+            string htmlContent = File.ReadAllText(templatePath);
+            foreach (ContentMetaData metaData in metaDataList)
+            {
+                htmlContent = htmlContent.Replace("{" + metaData.Placeholder + "}", metaData.Content);
```

Comment:
Use format string to build the string

```
// Old
htmlContent = htmlContent.Replace("{" + metaData.Placeholder + "}", metaData.Content);

// New
htmlContent = htmlContent.Replace($"{{{metaData.Placeholder}}}", metaData.Content);
```

Reply:
using string interpolation in the updated code
foreach (ContentMetaData metaData in metaDataList)
{
    htmlContent = htmlContent.Replace($"{{{metaData.Placeholder}}}", metaData.Content);
}